### DISCUSSION TITLE
How to design GNN to perform node regression on a single node?

### DISCUSSION THREAD

**Discussion by sunchaesk** [pyg-team/pytorch_geometric#7906] [Q&A]
Hi,
I am trying to build a model that a single value from the list of features. So for my graph, each node has multiple features, and for each node, I want the model to predict the next value for one of the feature values.

The thing is, for some of the nodes, the values are already determined so there is no need for the model to predict.

Currently, there is only 1 node that is of interest and there are 3 nodes connected to it where there is no need of a model.

I want to know if there is a way to disregard nodes from the training process?

The model right now has the structure as:
1. Graph Convolutional Layer
2. RNN
3. Linear layer with single out dimension
and the loss is calculated with the output and the single value from the single node of interest.

**Comment by rusty1s** [comment]
If you want to shrink your graph down in advance, you can use `utils.k_hop_subgraph` for this for example. Alternatively, you can take a look at `NeighborLoader`, but it might be overkill for your application.
